### ISSUE TITLE
Integrated Terminal command not found errors

### ISSUE THREAD

**Issue by ausiddiqui** [microsoft/vscode#124409]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: Code - Insiders 1.57.0-insider (4fd4e10e99eff9eb77408b71c7f58b7ed7458ee5, 2021-05-21T05:12:10.032Z)
- OS Version: Darwin x64 20.4.0

Steps to Reproduce:

On loading of the program the Integrated Terminal has issues loading and running scripts. Might be related to `$PATH` but not 100% sure. This is running on macOS.

```
/etc/zshrc:7: command not found: locale
/usr/local/Cellar/rbenv/1.1.2/libexec/rbenv-init: line 85: mkdir: command not found
/usr/local/bin/brew: line 16: readlink: command not found
/usr/local/bin/brew: line 17: dirname: command not found
/usr/local/bin/brew: line 16: readlink: command not found
/usr/local/bin/brew: line 17: dirname: command not found
/bin/bash: /usr/local/Library/Homebrew/brew.sh: No such file or directory
/usr/local/Cellar/rbenv/1.1.2/libexec/rbenv-init: line 85: mkdir: command not found
/Users/ash/.oh-my-zsh/oh-my-zsh.sh:69: command not found: rm
detect-clipboard:33: command not found: uname
(eval):116: command not found: dirname
(eval):116: command not found: dirname
__add_sys_prefix_to_path:6: command not found: dirname
__add_sys_prefix_to_path:7: command not found: dirname
```

Relevant settings.json options:
```
"terminal.integrated.defaultProfile.osx": "zsh",
"terminal.integrated.profiles.osx": {

        "bash": {
            "path": "bash",
            "icon": "terminal-bash"
        },
        "zsh": {
            "path": "zsh"
        },
        "fish": {
            "path": "fish"
        },
        "tmux": {
            "path": "tmux",
            "icon": "terminal-tmux"
        },
        "pwsh": {
            "path": "pwsh",
            "icon": "terminal-powershell"
        }
    },
```

My default shell is ZSH in `/bin/zsh` and is controlled using `~/.zshrc` file. All `export PATH` commands in the rc file are set without using quotes (saw something in SO that suggested removing quotes). The commands `which zsh` and `zsh --version` work in the integrated terminal but while `which brew` works, `brew -v` returns another `command not found` error.

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz (16 x 2300)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|32.00GB (4.84GB free)|
|Process Argv|--crash-reporter-id 19000199-b326-4074-8806-4977fba5c58e|
|Screen Reader|no|
|VM|0%|
</details>

**Comment by andyljones** [comment]
I think I might be suffering something similar here - does your $PATH in the integrated console look like your zshrc has been run twice?

**Comment by ausiddiqui** [comment]
Hey @andyljones how would I identify that?

**Comment by andyljones** [comment]
Try `echo $PATH` in the vscode terminal, compared to `echo $PATH` in the OS terminal

**Comment by ausiddiqui** [comment]
Oh right got it, thanks. Yeah they are different. These are missing in the VSCode `$PATH` variable. `<uname>` is just a place holder for the macOS account username. 

```
/Users/<uname>/miniconda3/bin:
/Users/<uname>/miniconda3/condabin:
/Users/<uname>/.rbenv/shims:
/usr/bin:
/usr/sbin:
/sbin:
/Library/Apple/usr/bin
```

When I rush `which zsh` I notice that VSCode Terminal says `bin/zsh` but in my regular terminal it is `/usr/local/bin/zsh`. Is there a way to verify what profile is being selected/active, when I do the command palette option Terminal: Select Default Profile, the list shows bash and zsh under profiles but not sure which one is active.

**Comment by vallentin** [comment]
Unrelated, but I had a similar issue on Windows with Git Bash. Where when I opened the terminal it would print "command not found" for a bunch of commands, I wasn't even able to execute `clear`.

In my case it was an issue with `PATH`, as `echo $PATH` would emit different results in VSCode vs outside VSCode. Opening VSCode through Git Bash, i.e. `code .` and the issue disappeared for that instance.

Not sure whether it was caused by a Windows or VSCode update, as both occurred at the same time. However, I resolved the issue by [upgrading Git, i.e. `git update-git-for-windows`](https://stackoverflow.com/questions/13790592/how-to-upgrade-git-on-windows-to-the-latest-version). (Reinstalling Git probably works as well.)

_Just leaving this comment, in case other people have the same issue on Windows, and find this thread._
